lib/roo_on_rails/logger.rb

```diff
@@ -0,0 +1,104 @@
+require 'logger'
+require 'delegate'
+require 'roo_on_rails/logfmt'
+
+module RooOnRails
+  # A compatible replacement for the standard Logger to provide context, similar
+  # to `ActiveSupport::TaggedLogging` but with key/value pairs in logfmt format.
+  #
+  #   logger = RooOnRails::Logger.new(STDOUT)
+  #   logger.with(a: 1, b: 2) { logger.info 'Stuff' }
+  #   # Logs "at=INFO msg=Stuff a=1 b=2"
+  #
+  #   logger.with(a: 1) { logger.with(b: 2) { logger.info('Stuff') } }
+  #   # Logs "at=INFO msg=Stuff a=1 b=2"
+  #
+  # The above methods persist the context in thread local storage so it will be
+  # attached to any logs made within the scope of the block, even in called
+  # methods. However, if your context only applies to the current log then you
+  # can chain off the `with` method.
+  #
+  #   logger.with(a: 1, b: 2).info('Stuff')
+  #   # Logs "at=INFO msg=Stuff a=1 b=2"
+  #
+  #   logger.with(a: 1) { logger.with(b: 2).info('Stuff')  }
+  #   # Logs "at=INFO msg=Stuff a=1 b=2"
+  #
+  # Hashes, arrays and any complex object that supports `#to_json` will be
+  # output in escaped JSON format so that it can be parsed out of the attribute
+  # values.
```

Comment:
According to the readme you can also do `logger.info('Stuff', a: 1, b: 2)` so might be worth adding that in here if you want to be comprehensive.

Reply:
According to the README... but not in practice (the current code does not support it). Do you think this is a feature we should add (back)?

Reply:
Actually, I can't see what you mention in the README :/
Happy to introduce this in a further PR though.